src/http/RequestHandler.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace markhuot\craftpest\http;
+
+use markhuot\craftpest\web\Application;
+use markhuot\craftpest\web\TestableResponse;
+use yii\web\HttpException;
+
+class RequestHandler
+{
+    private Application $app;
+
+    public function __construct(?Application $app = null)
+    {
+        $this->app = $app ?? \Craft::$app;
+    }
+
+    public function handle($request, $skipSpecialHandling = false): TestableResponse
+    {
+        $this->registerWithCraft($request);
+
+        try {
+
+            $this->app->trigger(Application::EVENT_BEFORE_REQUEST);
+
+            // The actual call
+            $response = $this->app->handleRequest($request, $skipSpecialHandling);
+            $response->prepare();
+
+            $this->app->trigger(Application::EVENT_AFTER_REQUEST);
+        }
+
+        // Catch any exceptions during handling
+        catch (\Exception $e) {
+
+            if (is_a($e, HttpException::class)) {
+                // Fake a response and set the HTTP status code
+                $response = \Craft::createObject(\markhuot\craftpest\web\TestableResponse::class);
+                $response->setStatusCode($e->statusCode);
+
+                // Error response
+                return $response;
+            }
+
+            // Something unexpected
+            echo $e->getMessage();
+            echo $e->getTraceAsString();
+            die;
+        }
+
+        return $response;
+    }
+
+    private function registerWithCraft($request): void
+    {
+        $this->app->getConfig()->getGeneral()->runQueueAutomatically = false;
+        $this->app->getView()->setTemplateMode($request->isCpRequest() ? 'cp' : 'site');
```

Comment:
Craft 3 support needs this to be a property, not a method.

```suggestion
        $this->app->getView()->setTemplateMode($request->isCpRequest ? 'cp' : 'site');
```